### DISCUSSION TITLE
Replacement for remark-variables?

### DISCUSSION THREAD

**Discussion by NickersWeb** [remarkjs/remark#1313] [Q&A] [answered]
I want to have custom-defined variables in my markdown. [remark-variables](https://github.com/mrzmmr/remark-variables) seems to already handle this. It is, however, no longer maintained. This was mentioned [here](https://github.com/orgs/remarkjs/discussions/1271).

I haven't been able to find a replacement for this library. Is there a replacement for this library?

I'm currently using remarkjs via [react-markdown](https://github.com/remarkjs/react-markdown). So preferably, a replacement library would also work with that library as well.

**Comment by wooorm** [comment, marked as answer]
You could use MDX’s expressions, or `remark-directives`’s components for this!

**Reply by NickersWeb** [reply]
That's good to hear. Could you link to these resources on how to use them? I really appreciate the help here.

**Reply by wooorm** [reply]
https://github.com/mdx-js/mdx - https://mdxjs.com
https://github.com/remarkjs/remark-directive
